File: app/src/main/java/client/sdk/myapplication/cirtag/data/ScannedProduct.kt
package client.sdk.myapplication.cirtag.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "scanned_products")
data class ScannedProduct(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val rawValue: String,
    val displayValue: String,
    val format: String,
    val type: String,
    val productName: String = "",
    val productDescription: String = "",
    val imageUrl: String = "",
    val productId: String = "",
    val price: String = "",
    val supplier: String = "",
    val skuId: String = "",
    val weight: String = "",
    val co2Total: String = "",
    val co2Details: String = "",       // "Raw Materials:3.15,Shipping & Transport:3.18,Manufacturing:33.81"
    val certifications: String = "",   // "ISO 14001,BPA Free,FCC Approved"
    val datasheetUrl: String = "",
    val scannedAt: Long = System.currentTimeMillis()
)

File: CirtagRN/src/types/ScannedProduct.ts
export interface ScannedProduct {
  id: number;
  rawValue: string;
  displayValue: string;
  format: string;
  type: string;
  productName: string;
  productDescription: string;
  imageUrl: string;
  productId: string;
  price: string;
  supplier: string;
  skuId: string;
  weight: string;
  co2Total: string;
  co2Details: string;       // "Raw Materials:3.15 Kg CO₂,Shipping & Transport:3.18 Kg CO₂"
  certifications: string;   // "ISO 14001,BPA Free,FCC Approved"
  datasheetUrl: string;
  documents: string;
  scannedAt: number;
}
